### PR TITLE
Changing stop targets to not error on non-existent items

### PR DESCRIPTION
**Pull request by insequent**
This is kind of a silly PR, and I totally understand if the Calico team is not interested. I was just getting annoyed with spam on the `stop-kubernetes-master` in particular. Especially when no volumes existed.

It'll still error on the explicit ones, but at least that's only 2 lines (one for docker, one for make)

### CODE REVIEW

**Comment by caseydavenport** [comment]
@insequent thanks for opening, and sorry for the delay on review - I've been out for a few weeks.

File: Makefile [hunk 1]
```diff
@@ -78,14 +78,14 @@ stop-kubernetes-master:
 	-docker rm -f calico-kubelet-master
 
 	# Remove any pods that the old kubelet may have started.
-	-docker rm -f $$(docker ps | grep k8s_ | awk '{print $$1}')
+	-docker ps -af name=k8s_ | awk 'NR < 2 {next}{print $$1}' | xargs -r docker rm -f
```
Comment (caseydavenport) [inline Makefile]: Hm, I think the use of `xargs -r` breaks this on BSD / Mac.  See here: http://stackoverflow.com/questions/8803987/what-is-the-equivalent-to-xargs-r-under-osx

I think this needs to be tweaked so the tests can successfully run on a Mac.

**Comment by insequent** [comment]
Thanks @caseydavenport. I hadn't tested on a Mac, but hopefully using the `while read` loop will work. Mind giving it a go? Also, I updated the other targets to have `##` comments, so `make help` is able to parse them correctly.

**Comment by insequent** [comment]
Missed comments for `clean` and rebased with master.

**Comment by caseydavenport** [comment]
OK - this worked for me on a Mac, and seems to be passing automated tests as well!

**Pull request merged**